packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -127,68 +127,95 @@ export class DefaultCrudRepository<
       `Entity ${entityClass.name} must have at least one id/pk property.`,
     );
 
-    this.modelClass = this.definePersistedModel(entityClass);
+    this.modelClass = this.definePersistedModel(entityClass, new Map());
   }
 
   // Create an internal legacy Model attached to the datasource
   private definePersistedModel(
     entityClass: typeof Model,
+    visited: Map<typeof Model, typeof juggler.PersistedModel>,
```

Comment:
Can we leverage DataSource's `ModelBuilder` instance that's already keeping track of models that have been defined, instead of creating a new map?

Reply:
Probably not as `ModelBuilder` has no idea of LB4 entity class.

Reply:
`ModelBuilder` is keeping track of all models that are backing LB4 entity classes. Whenever we create a new Repository for an LB4 entity class, a new (Persisted)Model is registered with the ModelBuilder behind the datasource.